src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,910 @@
+  +subdirectories: Map<string, TraceDirectoryTree>,
+|};
+
+type InstrumentsFrameInfo = {|
+  +key: string | number,
+  +name: string,
+  file?: string,
+  +line?: number,
+  +col?: number,
+|};
+
+type FormTemplateRunData = {|
+  +number: number,
+  +addressToFrameMap: Map<number, InstrumentsFrameInfo>,
+|};
+
+function readAsArrayBuffer(file: File): Promise<ArrayBuffer> {
+  return MaybeCompressedReader.fromFile(file).readAsArrayBuffer();
+}
+
+function readAsText(file: File): Promise<string> {
+  return MaybeCompressedReader.fromFile(file).readAsText();
+}
+
+function parseBinaryPlist(bytes) {
+  const text = 'bplist00';
+  for (let i = 0; i < 8; i++) {
+    if (bytes[i] !== text.charCodeAt(i)) {
+      throw new Error('File is not a binary plist');
+    }
+  }
+
+  return new BinaryPlistParser(
+    new DataView(bytes.buffer, bytes.byteOffset, bytes.byteLength)
+  ).parseRoot();
+}
+
+function decodeUTF8(bytes: Uint8Array): string {
+  const textDecoder = new TextDecoder();
+  if (bytes[bytes.length - 1] === 0) {
+    // Remove a single trailing null byte if present.
+    return textDecoder.decode(bytes.subarray(0, -1));
+  }
+  return textDecoder.decode(bytes);
+}
+
+function followUID(objects: any[], value: any): any {
+  return value instanceof UID ? objects[value.index] : value;
+}
+
+// This function constructs an object from given interpreter class and property list
+function patternMatchObjectiveC(
+  objects: any[],
+  value: any,
+  interpretClass: ($classname: string, obj: any) => any = x => x
+): any {
+  if (isDictionary(value) && value.$class) {
+    const name = followUID(objects, value.$class).$classname;
+    switch (name) {
+      case 'NSDecimalNumberPlaceholder': {
+        const length: number = value['NS.length'];
+        const exponent: number = value['NS.exponent'];
+        const byteOrder: number = value['NS.mantissa.bo'];
+        const negative: boolean = value['NS.negative'];
+        const mantissa = new Uint16Array(
+          new Uint8Array(value['NS.mantissa']).buffer
+        );
+        let decimal = 0;
+
+        for (let i = 0; i < length; i++) {
+          let digit = mantissa[i];
+
+          if (byteOrder !== 1) {
+            digit = ((digit & 0xff00) >> 8) | ((digit & 0x00ff) << 8);
+          }
+
+          decimal += digit * Math.pow(65536, i);
+        }
+
+        decimal *= Math.pow(10, exponent);
+        return negative ? -decimal : decimal;
+      }
+
+      // Replace NSData with a Uint8Array
+      case 'NSData':
+      case 'NSMutableData':
+        return value['NS.bytes'] || value['NS.data'];
+
+      // Replace NSString with a string
+      case 'NSString':
+      case 'NSMutableString':
+        return value['NS.bytes'] ? decodeUTF8(value['NS.bytes']) : '';
+
+      // Replace NSArray with an Array
+      case 'NSArray':
+      case 'NSMutableArray': {
+        if ('NS.objects' in value) {
+          return value['NS.objects'];
+        }
+        const array: any[] = [];
```

Comment:
please try to type this properly (same goes for other similar problems in this function)

Reply:
@julienw, Added the type.